diffrax/_brownian/tree.py

```diff
@@ -348,14 +401,21 @@ def _body_fun(_state: _State):
             _key = jnp.where(_cond, _key_st, _key_tu)
 
             _w = _split_interval(_cond, _w_stu, _w_inc)
-            _bkk = None
-            if self.levy_area is SpaceTimeLevyArea:
+
+            if self.levy_area is SpaceTimeTimeLevyArea:
+                assert _bkk_stu is not None and _bkk_st_tu is not None
+                _bkk = _split_interval(_cond, _bkk_stu, _bkk_st_tu)
+            else:
+                _bkk = None
+
+            if (
+                self.levy_area is SpaceTimeLevyArea
+                or self.levy_area is SpaceTimeTimeLevyArea
+            ):
                 assert _bhh_stu is not None and _bhh_st_tu is not None
                 _bhh = _split_interval(_cond, _bhh_stu, _bhh_st_tu)
-            elif self.levy_area is BrownianIncrement:
-                _bhh = None
             else:
-                assert False
+                _bhh = None
```

Comment:
I wonder if this can be more neatly expressed via a `jtu.tree_map(_split_interval, ...)`?

Not a strong feeling though.

Reply:
Does that improve compilation time? Otherwise I rather wouldn't, because it would require a very complicated `is_leaf=...`.